internal/ingress/controller/nginx.go

```diff
@@ -369,6 +384,7 @@ func (n *NGINXController) Start() {
 // Stop gracefully stops the NGINX master process.
 func (n *NGINXController) Stop() error {
 	n.isShuttingDown = true
+	time.Sleep(time.Duration(n.cfg.ShutdownGracePeriod) * time.Second)
```

Comment:
Before sleeping we might want to check if the shutdown is already happening. See about 7 lines below:

```
if n.syncQueue.IsShuttingDown() {
...
```

Reply:
Good point, I've moved the sleep below IsShuttingDown check. Thank you.